metrics/firebase/firestore/rules/firestore.rules

```diff
@@ -0,0 +1,68 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+    
+    match /projects/{projectId} {
+      allow read, delete: if isLoggedIn();
+      allow create, update: if isLoggedIn() && hasName();
```

Comment:
```suggestion
      allow create, update: if isLoggedIn() && projectIsValid();
```
to match build validation & make it more extensible.

Also maybe we need to change the naming of validation methods to conform style: isProjectValid - to match isLoggedIn?